crates/topos-p2p/src/behaviour/grpc/connection.rs

```diff
@@ -0,0 +1,90 @@
+use std::future::IntoFuture;
+
+use futures::{future::BoxFuture, FutureExt};
+use libp2p::{swarm::ConnectionId, Multiaddr};
+use tokio::sync::oneshot;
+use tonic::transport::Channel;
+
+use super::{
+    error::{OutboundConnectionError, OutboundError},
+    RequestId,
+};
+
+/// Connection struct which represent a connection between two nodes
+/// It contains the connection id, the address of the node, the request id
+/// and the gRPC channel which is used to communicate with the node.
+#[derive(Debug)]
+pub(crate) struct Connection {
+    /// The connection id
+    pub(crate) id: ConnectionId,
+
+    /// The address of the node
```

Comment:
THe address of the **remote** node right?

Reply:
It's the address of the node for this connection

Reply:
Hmm, so it's the address of whichever node dialed the other?

Reply:
Yes the address of the node represented by the connection